manager/controllers/app/blueprint_mgr.go

```diff
@@ -27,10 +26,15 @@ func containsTemplate(templateList []app.ComponentTemplate, moduleName string) b
 
 // GetBlueprintSignature returns the signature (name and namespace) of the blueprint owned by the given M4DApplication instance
 func (r *M4DApplicationReconciler) GetBlueprintSignature(applicationContext *app.M4DApplication) *app.Blueprint {
+	//sanity - app.Status.BlueprintNamespace should exist, since it represents the blueprint namespace name
+	if len(applicationContext.Status.BlueprintNamespace) == 0 {
+		return nil
+	}
```

Comment:
I had some issues understanding the flow and I suspect it has to do with this sanity check